core/api/src/main/java/com/blazebit/persistence/ConfigurationProperties.java

```diff
@@ -179,7 +179,19 @@
      * @see CTEBuilder#with(Class, CriteriaBuilder, boolean)
      */
     public static final String INLINE_CTES = "com.blazebit.persistence.inline_ctes";
-    
+
+    /**
+     * If set to true, the query plans are cached and reused.
+     * Valid values for this property are <code>true</code> and <code>false</code>.
+     * Default is <code>true</code>.
+     * This configuration option currently only takes effect when Hibernate is used as JPA provider.
+     *
+     * The property can be changed for a criteria builder before constructing a query.
+     *
+     * @since 1.5.0
+     */
+    public static final String QUERY_PLAN_CACHE_ENABLED = "com.blazebit.persistence.query_plan_cache_enabled";
```

Comment:
Isn't query caching opt-in anyway through `setCacheable(true)`?

Reply:
`setCacheable(true)` affects query result caching.

Reply:
Ah! My bad!